File: rust/14.最长公共前缀.rs
/*
 * @lc app=leetcode.cn id=14 lang=rust
 *
 * [14] 最长公共前缀
 */

// @lc code=start
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let mut common_prefix = String::new();
        let mut strs = strs;
        let s0 = strs.pop().unwrap();
        for i in 1..s0.len() {
            let prefix = String::from(s0.split_at(i).0);
            for s in &strs {
                if s.starts_with(&prefix) {
                    continue;
                } else {
                    return common_prefix;
                }
            }
            common_prefix = prefix.to_string();
        }
    
        for s in &strs {
            if s.starts_with(&s0) {
                continue;
            } else {
                return common_prefix;
            }
        }
        return s0.to_string();
    
    }
}
// @lc code=end

File: 14.最长公共前缀.go
/*
 * @lc app=leetcode.cn id=14 lang=golang
 *
 * [14] 最长公共前缀
 */
package leetcode

import "strings"

// @lc code=start
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	var sb strings.Builder
	for i := 0; ; i++ {
		for _, str := range strs {
			if i == len(str) || strs[0][i] != str[i] {
				return sb.String()
			}
		}
		sb.WriteByte(strs[0][i])
	}
}

// @lc code=end
